### ISSUE TITLE
helm: loki charts breaks when using prometheus serviceMonitor without grafana operator

### ISSUE THREAD

**Issue by mkoertgen** [grafana/loki#7546]
**Describe the bug**
The loki helm-chart does not support adding a prometheus ServiceMonitor without adding also grafana MetricsInstance.
This breaks in cases where users already use the prometheus operator but don't yet want to use the grafana operator which is still in beta. 

**To Reproduce**
1. Helm install with values like below

**Expected behavior**
The MetricsInstance-template should be opt-in the same as the grafana-operator (at least temporarily), e.g. by 
- testing for the CRD-capability (like the prometheus ServiceMonitor template) or
- testing for `monitoring.selfMonitoring.grafanaAgent.installOperator`
- ...

**Environment:**
 - Infrastructure: Kubernetes withouth Grafana Agent Operator CRDs
 - Deployment tool: helm, kustomize, flux

**Screenshots, Promtail config, or terminal output**

Minimal example for helm values
```yaml
loki:
  auth_enabled: false
  minio:
    enabled: false
monitoring:
  dashboards:
    enabled: true
    labels:
      grafana_dashboard: "1"
  rules:
    alerting: true
    enabled: true
  selfMonitoring:
    enabled: false # Disable, as this relies on Grafana Agent Operator
    grafanaAgent:
      installOperator: false # Disable operator for now (still beta)
    lokiCanary:
      enabled: false # Disable, as this relies on Grafana Agent Operator
  serviceMonitor:
    enabled: true # <--- !requires MetricsInstance CRD!
    # Setting explicitlly to null is a hard workaround for the fact that the loki chart does not support 
    # rendering a ServiceMonitor (prometheus-operator CRD) without a MetricsInstance (grafana-operator CRD).
    # NOTE: This will not work with kustomize-based tooling (e.g. flux) because kustomize will remove the null value.
    # metricsInstance: null 
read:
  persistence:
    size: 10Gi
    storageClass: ""
  replicas: 2
test:
  enabled: false # Disable, as this relies on Grafana Agent Operator
write:
  persistence:
    size: 10Gi
    storageClass: ""
  replicas: 2
```

**Comment by mkoertgen** [comment]
We don't want to to use the grafana agent operator yet. Our current workaround is to set `monitoring.serviceMonitor.enabled: false` and add the serviceMonitor separately.

**Comment by tyriis** [comment]
https://github.com/grafana/loki/issues/7529 at least related to this one

**Comment by mkoertgen** [comment]
Closed in favor of #7535 (duplicate)
